public/app/features/explore/TraceView/components/model/trace-viewer.ts

```diff
@@ -55,3 +55,34 @@ export const getTraceName = memoize(_getTraceNameImpl, (spans: TraceSpan[]) => {
   }
   return spans[0].traceID;
 });
+
+export function findSpanWithHeaderTags(spans: TraceSpan[]) {
```

Comment:
- [Jaeger](https://www.jaegertracing.io/docs/1.42/client-libraries/) clients have been retired, and OpenTelemetry SDKs should be used instead.
- Zipkin uses `http.method` at least (from Zipkin [swagger API](https://zipkin.io/zipkin-api/#/default/post_spans)), and `http.method`, `http.path` (from our [own docs](https://grafana.com/docs/grafana/latest/datasources/zipkin/)). I've added a `http.path` to our url const.
- This is not definitive proof however I think the general idea is to move towards OTEL. I've checked Tempo/Jaeger/Zipkin locally and Tempo on ops to see what attributes/resources (tags) we are using.

It only takes one of the tags to render the badges in the header. If no tags are present, no badges are rendered.

Not sure what extra value there is in using `kind: server` instead?

Reply:
Oh cool didn't know about the jeager deprecation.

Yeah probably makes sense as it is in that case.

> Not sure what extra value there is in using kind: server instead?

I mean probably does not make much difference as you already use `return tag.key === 'http.url' || tag.key === 'http.target';` here. Wass just wondering if there is any benefit or important difference depending on whether we get this data from first client span or first server span but seems like there should not be much difference otherwise.